src/test/ui/consts/const-mut-refs/const_mut_refs.stderr

```diff
@@ -0,0 +1,30 @@
+error[E0658]: references in constants may only refer to immutable values
+  --> $DIR/const_mut_refs.rs:31:17
+   |
+LL |     let _: [(); foo().bar()] = [(); 1];
```

Comment:
Since the return type of array length expressions is always a `usize`, we could allow `&mut` here. We probably don't need to worry about this now, however. Let's see if someone actually runs into this.

Reply:
We could do this in general. If the return type is guaranteed free of mutable references, we can allow them. But I agree, let's do it one step at a time.